etc/bin/run-functional-tests.sh

```diff
@@ -93,6 +93,11 @@ for ARG in "$@"; do
     esac
 done
 
+# ensure that TMPDIR is not a symlink
+# (we rely on path comparisons in many tests)
+TMPDIR="$(realpath "${TMPDIR}")"
+export TMPDIR
```

Comment:
Failing for more on Ubuntu:

```
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ env | grep TMP

```

And running one of the tests that failed on Travis

```
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ ./etc/bin/run-functional-tests.sh ./tests/events/18-suite-event-mail.t 
./etc/bin/run-functional-tests.sh: line 98: TMPDIR: unbound variable
[16:28:10] ./tests/events/18-suite-event-mail.t .. skipped: "mail" command not available
[16:28:10]
Files=1, Tests=0,  1 wallclock secs ( 0.02 usr  0.01 sys +  0.90 cusr  0.08 csys =  1.01 CPU)
Result: NOTESTS
(venv) kinow@ranma:~/Development/python/workspace/cylc-flow$ echo $?
0
```

But I suspect on Travis as the test is not skipped, it is probably failing? In Ubuntu's `mktemp` docs say:

>  -p DIR, --tmpdir[=DIR]  interpret TEMPLATE relative to DIR; if DIR is not
>                        specified, use $TMPDIR if set, else /tmp.

So I guess there should be some way in Shell to say `realpath"${TMPDIR-or /tmp if not set?}"` (i.e. use `/tmp` if no `TMPDIR` variable? not sure if the best solution)

Reply:
Yep can do, I thought we had issues with tests failing where TMPDIR wasn't set, perhaps not.

Reply:
Also this appears to have broken all mail tests!